File: TimeTravel/src/main/java/ajc/sopra/TimeTravel/model/Machine.java
package ajc.sopra.TimeTravel.model;

import java.time.LocalDate;
import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("machine")

public class Machine {
	@JsonView(JsonViews.Common.class)
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	@JsonView(JsonViews.Common.class)
	@Column(name="DateMachine",nullable = false)
	private LocalDate dateMachine;
	@JsonView(JsonViews.Common.class)
	@Column(nullable = false)
	@Enumerated
	private TypeMachine typeMachine;
	@JsonView(JsonViews.Common.class)
	@Column(nullable = false)
	@Enumerated
	private EtatMachine etatMachine;
	
	public Machine(Integer id, LocalDate dateMachine, TypeMachine typeMachine, EtatMachine etatMachine) {
		this.id = id;
		this.dateMachine = dateMachine;
		this.typeMachine = typeMachine;
		this.etatMachine = etatMachine;
	}
	
	public Machine(LocalDate dateMachine, TypeMachine typeMachine, EtatMachine etatMachine) {
		this.dateMachine = dateMachine;
		this.typeMachine = typeMachine;
		this.etatMachine = etatMachine;
	}
	
	public Machine() {}
	
	
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public LocalDate getDateMachine() {
		return dateMachine;
	}
	public void setDateMachine(LocalDate dateMachine) {
		this.dateMachine = dateMachine;
	}

	public TypeMachine getTypeMachine() {
		return typeMachine;
	}

	public void setTypeMachine(TypeMachine typeMachine) {
		this.typeMachine = typeMachine;
	}

	public EtatMachine getEtatMachine() {
		return etatMachine;
	}

	public void setEtatMachine(EtatMachine etatMachine) {
		this.etatMachine = etatMachine;
	}

	@Override
	public int hashCode() {
		return Objects.hash(etatMachine);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Machine other = (Machine) obj;
		return etatMachine == other.etatMachine;
	}
	
	

}

File: TimeTravelFront/src/app/timetravel/model/machine.ts
import { EtatMachine } from "../enum/etat-machine";
import { TypeMachine } from "../enum/type-machine";

export class Machine {
  public get etatMachine(): EtatMachine | undefined {
    return this._etatMachine;
  }
  public set etatMachine(value: EtatMachine | undefined) {
    this._etatMachine = value;
  }
  public get typeMachine(): TypeMachine | undefined {
    return this._typeMachine;
  }
  public set typeMachine(value: TypeMachine | undefined) {
    this._typeMachine = value;
  }
  public get dateMachine(): Date | undefined {
    return this._dateMachine;
  }
  public set dateMachine(value: Date | undefined) {
    this._dateMachine = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }
  constructor(
    private _id?: number,
    private _dateMachine?: Date,
    private _typeMachine?: TypeMachine,
    private _etatMachine?: EtatMachine
  ) {}
}
